src/compiler/checker.ts

```diff
@@ -100,11 +100,18 @@ module ts {
         let globalIterableType: ObjectType;
 
         let anyArrayType: Type;
+        let globalTypedPropertyDescriptorType: ObjectType;
+        let globalClassDecoratorType: ObjectType;
+        let globalClassAnnotationType: ObjectType;

```

Comment:
let's try to limit the number of concepts users have to deal with and keep the decorator only falvor.